framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -544,16 +544,46 @@ void MediaPlayerImpl::playback()
 	ssize_t num_read = mInputDataSource->read(mBuffer,(int)mBufSize);
 	meddbg("num_read : %d\n", num_read);
 	if (num_read > 0) {
-		start_audio_stream_out(mBuffer, get_output_bytes_to_frame((unsigned int)num_read));
+		int ret = start_audio_stream_out(mBuffer, get_output_bytes_to_frame((unsigned int)num_read));
+		if (ret < 0) {
+			PlayerWorker& mpw = PlayerWorker::getWorker();
+			switch (ret) {
+			case AUDIO_MANAGER_XRUN_STATE:
+				meddbg("AUDIO_MANAGER_XRUN_STATE\n");
+				break;
+			default:
+				meddbg("audio manager error : %d\n", ret);
+				notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+				mpw.enQueue(&MediaPlayerImpl::stopByError, shared_from_this());;
+				break;
+			}
+		}
 	} else if (num_read == 0) {
 		notifyObserver(PLAYER_OBSERVER_COMMAND_FINISHIED);
 		stop();
 	} else {
+		PlayerWorker& mpw = PlayerWorker::getWorker();
+		meddbg("InputDatasource read error\n");
 		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
-		stop();
+		mpw.enQueue(&MediaPlayerImpl::stopByError, shared_from_this());;
 	}
 }
 
+void MediaPlayerImpl::stopByError()
+{
+	PlayerWorker& mpw = PlayerWorker::getWorker();
+
+	meddbg("PlayerWorker Worker : stopOnPlaybackByError\n");
+
+	audio_manager_result_t result = stop_audio_stream_out();
+	if (result != AUDIO_MANAGER_SUCCESS) {
+		meddbg("stop_audio_stream_out failed ret : %d\n", result);
```

Comment:
Need to add TODO comment in order to consider the stop failure case in future.

Reply:
Is there something to do if stop_audio_stream_out failed?